src/components/PopoverWithMeasuredContent.js

```diff
@@ -59,138 +60,118 @@ const defaultProps = {
+    const popoverHeight = useRef(props.popoverDimensions.height);
+    const [isContentMeasured, setIsContentMeasured] = useState(popoverWidth.current > 0 && popoverHeight.current > 0);
+    const [isVisible, setIsVisible] = useState(false);
 
     /**
      * When Popover becomes visible, we need to recalculate the Dimensions.
      * Skip render on Popover until recalculations have done by setting isContentMeasured false as early as possible.
-     *
-     * @static
-     * @param {Object} props
-     * @param {Object} state
-     * @return {Object|null}
      */
-    static getDerivedStateFromProps(props, state) {
+    useEffect(() => {
         // When Popover is shown recalculate
-        if (!state.isVisible && props.isVisible) {
-            return {isContentMeasured: lodashGet(props, 'popoverDimensions.width', 0) > 0 && lodashGet(props, 'popoverDimensions.height', 0) > 0, isVisible: true};
+        if (!isVisible && props.isVisible) {
+            // We use additional function to guarantee that async state change would be completed
+            setIsVisible(() => {
+                setIsContentMeasured(lodashGet(props, 'popoverDimensions.width', 0) > 0 && lodashGet(props, 'popoverDimensions.height', 0) > 0);
+                return true;
+            });
         }
         if (!props.isVisible) {
-            return {isVisible: false};
-        }
-        return null;
-    }
-
-    shouldComponentUpdate(nextProps, nextState) {
-        if (this.props.isVisible && (nextProps.windowWidth !== this.props.windowWidth || nextProps.windowHeight !== this.props.windowHeight)) {
-            return true;
+            setIsVisible(false);
         }
-
-        // This component does not require re-render until any prop or state changes as we get the necessary info
-        // at first render. This component is attached to each message on the Chat list thus we prevent its re-renders
-        return !_.isEqual(_.omit(this.props, ['windowWidth', 'windowHeight']), _.omit(nextProps, ['windowWidth', 'windowHeight'])) || !_.isEqual(this.state, nextState);
-    }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [props.popoverDimensions.width, props.popoverDimensions.height, props.isVisible, isVisible]);
 
     /**
      * Measure the size of the popover's content.
      *
      * @param {Object} nativeEvent
      */
-    measurePopover({nativeEvent}) {
-        this.popoverWidth = nativeEvent.layout.width;
-        this.popoverHeight = nativeEvent.layout.height;
-        this.setState({isContentMeasured: true});
-    }
+    const measurePopover = ({nativeEvent}) => {
+        popoverWidth.current = nativeEvent.layout.width;
+        popoverHeight.current = nativeEvent.layout.height;
+        setIsContentMeasured(true);
+    };
 
-    /**
-     * Calculate the adjusted position of the popover.
-     *
-     * @returns {Object}
-     */
-    calculateAdjustedAnchorPosition() {
+    const adjustedAnchorPosition = useMemo(() => {
```

Comment:
The use of `useMemo` here won't work well with `popoverWidth` and `popoverHeight`. I think we will have to convert those back to state in order to be able to add them as `useMemo` dependencies.